core/templates/pages/library-page/search-bar/search-bar.component.spec.ts

```diff
@@ -0,0 +1,239 @@
+    spyOnProperty(
+      classroomBackendApiService,
+      'onInitializeTranslation').and.returnValue(initTranslationEmitter);
+    spyOnProperty(
+      i18nLanguageCodeService,
+      'onPreferredLanguageCodesLoaded').and.returnValue(
+      preferredLanguageCodesLoadedEmitter);
+    ctrl.$onInit();
+  }));
+
+  afterEach(function() {
+    ctrl.$onDestroy();
+  });
+
+  it('should initialize controller properties after its initialization',
+    function() {
+      expect(Object.keys(ctrl.selectionDetails)).toContain('categories');
+      expect(Object.keys(ctrl.selectionDetails)).toContain('languageCodes');
+
+      expect(ctrl.translationData).toEqual({
+        categoriesCount: 0,
+        languagesCount: 0
+      });
+    });
+
+  it('should get placeholder and button text translation when translation' +
+    ' is initialized', function() {
+    initTranslationEmitter.emit();
+
+    expect(ctrl.searchBarPlaceholder).toBe('I18N_LIBRARY_SEARCH_PLACEHOLDER');
+    expect(ctrl.categoryButtonText).toBe('I18N_LIBRARY_ALL_CATEGORIES');
+    expect(ctrl.languageButtonText).toBe('I18N_LIBRARY_ALL_LANGUAGES');
+  });
+
+  it('should search for all content in given languages', function() {
+    expect(ctrl.isSearchInProgress()).toBe(false);
+
+    mockWindow.location.pathname = '/search/find';
+    preferredLanguageCodesLoadedEmitter.emit(['en', 'es', 'hi']);
+    expect(ctrl.isSearchInProgress()).toBe(true);
+
+    expect(ctrl.translationData).toEqual({
+      categoriesCount: 0,
+      languagesCount: 3
+    });
+    expect($location.url()).toBe(
+      '/find?q=&language_code=(%22en%22%20OR%20%22es%22%20OR%20%22hi%22)');
+
+    preferredLanguageCodesLoadedEmitter.emit(['en']);
+
+    expect(ctrl.translationData).toEqual({
+      categoriesCount: 0,
+      languagesCount: 2
+    });
+    expect($location.url()).toBe(
+      '/find?q=&language_code=(%22es%22%20OR%20%22hi%22)');
+  });
+
+  it('should filter and search content by categories, language and text',
+    function() {
+      var calls = 0;
+      spyOn(searchService, 'getCurrentUrlQueryString').and.callFake(function() {
+        calls++;
+        if (calls === 1) {
+          return (
+            '?q=%22sun%22&language_code=(%22pt%22)&category=(%22astronomy%22)');
+        } else if (calls === 2) {
+          return '?q=%22sun%22';
+        }
+      });
+      mockWindow.location.search = (
+        '?q=%22sun%22&language_code=(%22pt%22)&category=(%22astronomy%22)');
+      spyOn(urlService, 'getUrlParams').and.returnValue({q: ''});
+      $scope.$digest();
+      preferredLanguageCodesLoadedEmitter.emit([]);
+
+      expect(mockWindow.location.href).toBe(
+        '/search/find?q=%22sun%22&category=("astronomy")&language_code=("pt")');
+    });
+
+  it('should filter and search content by categories, language and text' +
+    ' when url location changes', function() {
+    var calls = 0;
+    spyOn(searchService, 'getCurrentUrlQueryString').and.callFake(function() {
```

Comment:
Ditto

Reply:
I've tried to test this function but `getCurrentUrlQueryString` was always returning the same value. It must return different values to test the search-bar behavior so that's the way I've found to do it.